### DISCUSSION TITLE
How to update to DSM 7.2.1-69057 Update 1 with already installed NVMe

### DISCUSSION THREAD

**Discussion by nounours18200** [007revad/Synology_enable_M2_volume#130] [Q&A]
Hello,

I have successfully installed two NVMe on my DS1821+ (under DSM 7.2-64570 Update 3) using the script "synology_enable_M2_volume": a new Storage Pool has been created using the Storage Manager and works perfectly ! 
I have run ONLY this script, **without** running the "Synology_HDD_db" script.

But a new DSM version has been released a week ago: version 7.2.1-69057 Update 1

I have read that **"After any DSM update you will need to run this script, and the Synology_HDD_db script again"**, but if I do this after having updated DSM to 7.2.1-69057 Update 1, I suppose that ALL the data will be deleted by the "synology_enable_M2_volume" isn't it ? 

In addition, I read that there is an option to prevent the HDD database update in the "Synology_HDD_db" script, but if I do this I fear that the DSM update crashes because it cannot update the database as it wants to...

Please tell me how to proceed safely ?

Thank you ,

**Comment by 007revad** [comment]
That **"After any DSM update you will need to run this script, and the Synology_HDD_db script again"** should really say:
-  "After any DSM update you will need to run this script again"
-  "If you also ran the Synology_HDD_db script you will also need to run it again"

If just running Synology_enable_M2_volume did all you needed then there's no need to also run Synology_HDD_db.

Personally I only run the Synology_HDD_db script on my DS1821+ (as it achieves the same result as Synology_enable_M2_volume on a DS1821+).

Synology_HDD_db runs after DSM has updated so the script won't prevent a DSM update from installing a new version of the drive database. 

Synology_HDD_db only prevents DSM downloading a newer version of the drive database when you replace a drive. Any time you install a new drive DSM checks online for a new database version.

**Reply by nounours18200** [reply]
Thank you for your reply 007revad, 

but if I run the "Synology_enable_M2_volume" script after a DSM update, will the script erase ALL my data in the NVMes by formating them ? or will it keep the NVMe storage pool unchanged ?.

So will DSM still see the Storage Pool on the NVMes after the DSM update ? or do I have to recreate the Storage Pool and restore all the data that were on the NVMe before the DSM update ?

Thank you very much, and I apologize if my question seems obvious for you !

**Reply by 007revad** [reply]
The Synology_enable_M2_volume script doesn't touch your NVMe drives. It just changes a file in DSM.

Now there is also a different script with a similar name, Synology_M2_volume, which does delete what's on the NVMe drive(s) but you have to answer yes at a number of warnings so you can't accidentally run that script and delete your NVMe volume.

**Reply by nounours18200** [reply]
Very good, so if I understand correctly:

1-.I update DSM
2-.I run the "synology_enable_M2_volume" script
3-.I run the "synology_hdd_db" script (but I don't have run it previously: I had just run "synology_enable_m2_volume" and this allowed me to create a new NVMe storage pool via DSM Storage Manager).

and my data and VMs located on the NVMe should still exist (?).

**Reply by 007revad** [reply]
You don't even need the "synology_enable_M2_volume" script if you run "synology_hdd_db".

Just [schedule "synology_hdd_db" to run when the Synology boots](https://github.com/007revad/Synology_HDD_db/blob/main/how_to_schedule.md). 

Then when you update DSM and it reboots "synology_hdd_db" will run automatically, and all your data and VMs located on the NVMe will still exist.

**Reply by nounours18200** [reply]
I thank you VERY much for your reply and your support: I will perform this to run the script automatically (I currently configure my new PC, so i will do it a bit later...).
Your scripts are definitely a "must have": congratulations for your work !

**Reply by oriddlero** [reply]
Did you do it? I'm in the same boat and i have lots of critical packages running on my RAID1 m.2 array.  It's backed up of course but I would rather not have a freak out project to deal with if it does not pop right back up after the script.  Tell me how it went please :)

**Reply by nounours18200** [reply]
Hi oriddlero,
Yes I did: it worked absolutely perfect.
The NAS has performed the DSM update, and after its restart, all my storage pools and data were exactly as before, including of course the M2 storage pool.

**Reply by oriddlero** [reply]
Thank you for the prompt response, are you saying you just ran the DSM update and you did not even need to re-run the 'synology_enable_M2_volume' script? Or are you saying you re-ran the script and the m.2 volume just popped back up and everything was back the way it was before the update?

what model are you running? I'm on DS1621xs+  I'm dreading upgrading DSM considering my complex configuration/relationship between the M.2 volume and mag array.

**Reply by nounours18200** [reply]
Hello,

I am using the DS-1821+ in a SHR-2 configuration, with 32GB of RAM (not Synology but Crucial) and 2 x 2TB NVMe Samsung 980 Pro (in a JBOD configuration --> 4TB) used for the Virtual Machines.

BEFORE updating DSM, you MUST copy the revad007 script to a specific shared folder of your NAS (I have named mine "scripts" as recommended): just follow the revad007 instructions here: https://github.com/007revad/Synology_HDD_db/blob/main/how_to_schedule.md

When you have copied the script AND put it in the task scheduler with the revad007 recommended parameters, the script will be run automatically each time the NAS reboots. So you don't have to execute anything : just let the NAS restart after a DSM update, and after the restart is completed, you will see all your storage pools as they were prior the update.

When you have implemented the script in the task scheduler with the revad007 parameters, your Synology is as it should always have been, i.e. working with non-Synology branded extensions, like the QNAPs are. My next NAS will probably be a QNAP just because of this unacceptable Synology approach regarding third party parts.

You just have to play golf instead of worry about your NAS !

Do not hesitate to ask if you need,

**Reply by oriddlero** [reply]
Thanks again for the response.   I know it's a simple procedure, but it just makes me feel better to re-confirm the steps and the expected outcome.   Maybe I will do this some point this week.

**Reply by oriddlero** [reply]
Finally grew the balls to do this today.  Scheduled the database script to run on boot just in case Everything went smooth and perfect on DS1621xs+ w/ DSM 7.2.1-69057 Update 5 -- TY =]
